### DISCUSSION TITLE
Key exchange certificate check failure with hostname mismatch

### DISCUSSION THREAD

**Discussion by Reginard** [iterate-ch/cyberduck#16260] [Q&A] [answered]
I'm suddenly seeing this error when trying to sftp to my Proxmox servers.

```
KeyExchange certificate check failed: Hostname `10.x.x.x` doesn't match any of the principals: `hostname`, `hostname.home.arpa`, `10.x.x.x (old IP address)`, `ipv6 address`. The connection attempt was rejected. The server may be down, or your network may not be properly configured.
```

The servers are up and I can SSH to them from terminal. I have removed and reinstalled by still not working.

macOS Sonoma 14.6.1 (23G93)
Cyberduck Version 9.0.1 (41941)

Anyone seen this before?

**Comment by dkocher** [comment]
Bug report in #15354.

**Comment by dkocher** [comment]
Cyberduck verifies host key certificates by default. If your server is setup to use host key certificates for key exchange. Make sure the hostname configured in the bookmark matches the server name in the certificate.

**Reply by AliveDevil** [reply]
Also verify, that the ~/.ssh/known_hosts file doesn't have conflicting entries.

**Comment by Reginard** [comment]
I have verified ~/ssh/knownhosts by removing all entries. I've also checked Keychain and can confirm that the entries do not exist. The issue is still happening but never used to. I may try an older version of Cyberduck if I can find one.

**Reply by Reginard** [reply]
Followed the first post in this article [https://forum.proxmox.com/threads/obsolete-ssh-host-key-certificates-how-to-bypass-ssh-known_hosts-bug-s.137809/](url) and now feel I'm on the right track although the key exchange still fails

**Comment by Reginard** [comment, marked as answer]
Resolved by using fully qualified domain name (FQDN) instead of IP address
